### PR TITLE
opencv migrations

### PR DESCRIPTION
**Pull request by fxcoudert**
Starting again on opencv migrations.

Open issues that need to be dealt with:

- opencv with python3: https://github.com/Homebrew/homebrew-science/issues/5463
- opencv java link issue: https://github.com/Homebrew/homebrew-science/issues/3268

(other issues are old, closed, CUDA-related or do not provide enough information to act upon)

### CODE REVIEW

**Comment by fxcoudert** [comment]
poke @iMichka

File: Formula/opencv.rb [hunk 1]
```diff
@@ -0,0 +1,158 @@
+class Opencv3 < Formula
```
Comment (ilovezfs) [inline Formula/opencv.rb]: class should be opencv

Comment (iMichka) [inline Formula/opencv.rb]: And the other one misses AT2

File: Formula/opencv.rb [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  desc "Open source computer vision library, version 3"
+  homepage "http://opencv.org/"
+  revision 1
```
Comment (ilovezfs) [inline Formula/opencv.rb]: not needed since it's an upgrade 2.4.13.2 -> 3.2.0

File: Formula/opencv@2.rb [hunk 3]
```diff
@@ -0,0 +1,98 @@
+class Opencv < Formula
```
Comment (ilovezfs) [inline Formula/opencv@2.rb]: class needs to be opencvat2

File: Formula/opencv.rb [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  depends_on "libpng"
+  depends_on "libtiff"
+  depends_on "openexr"
+  depends_on :python => :recommended unless MacOS.version > :snow_leopard
+  depends_on "ffmpeg" => :optional
+  depends_on "tbb" => :optional
+  depends_on :java => :optional
+  depends_on :python3 => :optional
+
+  with_python = build.with?("python") || build.with?("python3")
```
Comment (ilovezfs) [inline Formula/opencv.rb]: numpy only has "without" options now so no options should be needed

File: Formula/opencv@2.rb [hunk 5]
```diff
@@ -0,0 +1,98 @@
+class OpencvAT2 < Formula
+  desc "Open source computer vision library"
+  homepage "http://opencv.org/"
+  url "https://github.com/opencv/opencv/archive/2.4.13.2.tar.gz"
+  sha256 "4b00c110e6c54943cbbb7cf0d35c5bc148133ab2095ee4aaa0ac0a4f67c58080"
+  revision 1
```
Comment (ilovezfs) [inline Formula/opencv@2.rb]: this is technically a new formula. we're just going to delete homebrew/science/opencv not migrate it

**Comment by iMichka** [comment]
I think the python3 thing is solved by this PR that got abandoned: https://github.com/Homebrew/brew/pull/2499

It tells you to setup your path and stuff to make it work.
But I never really tested all this myself, but once we have a clean build here I can give it a shot locally.

File: Formula/opencv.rb [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+
+  # Dependencies use fortran, which leads to spurious messages about gcc
+  cxxstdlib_check :skip
+
+  resource "icv-macosx" do
+    url "https://raw.githubusercontent.com/opencv/opencv_3rdparty/81a676001ca8075ada498583e4166079e5744668/ippicv/ippicv_macosx_20151201.tgz", :using => :nounzip
+    sha256 "8a067e3e026195ea3ee5cda836f25231abb95b82b7aa25f0d585dc27b06c3630"
+  end
+
+  def arg_switch(opt)
```
Comment (ilovezfs) [inline Formula/opencv.rb]: this is not idiomatic in core

File: Formula/opencv@2.rb [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+  depends_on "libtiff"
+  depends_on "openexr"
+  depends_on :python => :recommended if MacOS.version <= :snow_leopard
+  depends_on "numpy" => :recommended if build.with? "python"
+  depends_on "ffmpeg" => :optional
+  depends_on "tbb" => :optional
+  depends_on :java => :optional
+  depends_on :ant if build.with? "java"
+
+  def arg_switch(opt)
```
Comment (ilovezfs) [inline Formula/opencv@2.rb]: this is not idiomatic in core

File: Formula/opencv.rb [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+  option "with-contrib", 'Build "extra" contributed modules'
+  option "with-java", "Build with Java support"
+  option "with-tbb", "Enable parallel code in OpenCV using Intel TBB"
+  option "without-numpy", "Use a numpy you've installed yourself instead of a Homebrew-packaged numpy"
+  option "without-python", "Build without Python support"
+
+  option :cxx11
+
+  depends_on :ant => :build if build.with? "java"
```
Comment (ilovezfs) [inline Formula/opencv.rb]: the java option doesn't exist yet up here. it has to be below `depends_on :java => :optional`

**Comment by iMichka** [comment]
Do we want to consider the update to version 3.3.0 or are we doing this in a separate step? https://github.com/Homebrew/homebrew-science/pull/6078

It removes a bunch of code, which is nice.

File: Formula/opencv@2.rb [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+  depends_on "jpeg"
+  depends_on "libpng"
+  depends_on "libtiff"
+  depends_on "openexr"
+  depends_on :python => :recommended if MacOS.version <= :snow_leopard
+  depends_on "numpy" => :recommended if build.with? "python"
+  depends_on "ffmpeg" => :optional
+  depends_on "tbb" => :optional
+  depends_on :java => :optional
+  depends_on :ant if build.with? "java"
```
Comment (ilovezfs) [inline Formula/opencv@2.rb]: should be marked as `=> :build`

File: Formula/opencv.rb [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+
+    resource "contrib" do
+      url "https://github.com/opencv/opencv_contrib.git"
+    end
+  end
+
+  option "with-contrib", 'Build "extra" contributed modules'
+  option "with-java", "Build with Java support"
+  option "with-tbb", "Enable parallel code in OpenCV using Intel TBB"
+  option "without-numpy", "Use a numpy you've installed yourself instead of a Homebrew-packaged numpy"
```
Comment (ilovezfs) [inline Formula/opencv.rb]: I don't think we need this

File: Formula/opencv.rb [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+  option "without-python", "Build without Python support"
```
Comment (ilovezfs) [inline Formula/opencv.rb]: That's already the default message

File: Formula/opencv.rb [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+
+  option :cxx11
```
Comment (ilovezfs) [inline Formula/opencv.rb]: can we just make this the default?

File: Formula/opencv.rb [hunk 13]
```diff
@@ -0,0 +1,157 @@
...
+  depends_on "eigen"
+  depends_on "jpeg"
+  depends_on "libpng"
+  depends_on "libtiff"
+  depends_on "openexr"
+  depends_on :python => :recommended unless MacOS.version > :snow_leopard
+  depends_on "ffmpeg" => :optional
+  depends_on "tbb" => :optional
+  depends_on :java => :optional
+  depends_on :python3 => :optional
```
Comment (ilovezfs) [inline Formula/opencv.rb]: can we build with support for both?

File: Formula/opencv.rb [hunk 14]
```diff
@@ -0,0 +1,157 @@
...
+      py_prefix = `python-config --prefix`.chomp
+      py_lib = "#{py_prefix}/lib"
+      args << "-DPYTHON2_EXECUTABLE=#{which "python"}"
+      args << "-DPYTHON2_LIBRARY=#{py_lib}/libpython2.7.dylib"
+      args << "-DPYTHON2_INCLUDE_DIR=#{py_prefix}/include/python2.7"
+    end
+
+    if build.with? "python3"
+      # Reset PYTHONPATH, workaround for https://github.com/Homebrew/homebrew-science/pull/4885
+      ENV["PYTHONPATH"] = ""
```
Comment (ilovezfs) [inline Formula/opencv.rb]: I'd probably just put ENV.delete("PYTHONPATH") at the beginning of install and leave it at that.

File: Formula/opencv.rb [hunk 15]
```diff
@@ -0,0 +1,157 @@
...
+
+    if ENV.compiler == :clang && !build.bottle?
+      args << "-DENABLE_SSSE3=ON" if Hardware::CPU.ssse3?
+      args << "-DENABLE_SSE41=ON" if Hardware::CPU.sse4?
+      args << "-DENABLE_SSE42=ON" if Hardware::CPU.sse4_2?
+      args << "-DENABLE_AVX=ON" if Hardware::CPU.avx?
+      args << "-DENABLE_AVX2=ON" if Hardware::CPU.avx2?
+    end
+
+    inreplace buildpath/"3rdparty/ippicv/downloader.cmake",
```
Comment (ilovezfs) [inline Formula/opencv.rb]: what's going on here? has this been reported?

File: Formula/opencv.rb [hunk 16]
```diff
@@ -0,0 +1,157 @@
...
+      "${OPENCV_ICV_PLATFORM}-${OPENCV_ICV_PACKAGE_HASH}",
+      "${OPENCV_ICV_PLATFORM}"
+    resource("icv-macosx").stage buildpath/"3rdparty/ippicv/downloads/macosx"
+
+    mkdir "macbuild" do
```
Comment (ilovezfs) [inline Formula/opencv.rb]: build

File: Formula/opencv.rb [hunk 17]
```diff
@@ -0,0 +1,157 @@
...
+      system "cmake", "..", *args
+      system "make"
```
Comment (ilovezfs) [inline Formula/opencv.rb]: does just make install fail?

**Comment by ilovezfs** [comment]
we should add a formula rename for opencv3 to opencv

File: Formula/opencv.rb [hunk 18]
```diff
@@ -0,0 +1,157 @@
+class Opencv < Formula
+  desc "Open source computer vision library, version 3"
+  homepage "http://opencv.org/"
```
Comment (ilovezfs) [inline Formula/opencv.rb]: actually does need revision 1 if we're going to do the formula rename from opencv3

File: Formula/opencv.rb [hunk 19]
```diff
@@ -0,0 +1,157 @@
...
+    url "https://github.com/opencv/opencv/archive/3.2.0.tar.gz"
+    sha256 "b9d62dfffb8130d59d587627703d5f3e6252dce4a94c1955784998da7a39dd35"
+
+    resource "contrib" do
+      url "https://github.com/opencv/opencv_contrib/archive/3.2.0.tar.gz"
+      sha256 "1e2bb6c9a41c602904cc7df3f8fb8f98363a88ea564f2a087240483426bf8cbe"
+    end
+  end
+
+  head do
```
Comment (ilovezfs) [inline Formula/opencv.rb]: I think we have enough problems with these formulae not to need head.

File: Formula/opencv.rb [hunk 20]
```diff
@@ -0,0 +1,157 @@
...
+    url "https://github.com/opencv/opencv.git"
+
+    resource "contrib" do
+      url "https://github.com/opencv/opencv_contrib.git"
+    end
+  end
+
+  option "with-contrib", 'Build "extra" contributed modules'
```
Comment (ilovezfs) [inline Formula/opencv.rb]: feels out of scope since quality control is going to be low

File: Formula/opencv@2.rb [hunk 21]
```diff
@@ -0,0 +1,98 @@
...
+    end
+
+    if ENV.compiler == :clang && !build.bottle?
+      args << "-DENABLE_SSSE3=ON" if Hardware::CPU.ssse3?
+      args << "-DENABLE_SSE41=ON" if Hardware::CPU.sse4?
+      args << "-DENABLE_SSE42=ON" if Hardware::CPU.sse4_2?
+      args << "-DENABLE_AVX=ON" if Hardware::CPU.avx?
+    end
+
+    mkdir "macbuild" do
```
Comment (ilovezfs) [inline Formula/opencv@2.rb]: build

File: Formula/opencv@2.rb [hunk 22]
```diff
@@ -0,0 +1,98 @@
...
+      system "cmake", "..", *args
+      system "make"
```
Comment (ilovezfs) [inline Formula/opencv@2.rb]: does just make install fail

File: Formula/opencv.rb (hunk 19 above)
Comment (iMichka) [inline Formula/opencv.rb]: There are some websites/blog posts out there recommending to build opencv with `--head`. Never understood why. But it allowed me to close tens issues just based on the fact that it was head. So +1 on removing it.

File: Formula/opencv.rb [hunk 23]
```diff
@@ -0,0 +1,157 @@
...
+  depends_on "ffmpeg" => :optional
+  depends_on "tbb" => :optional
+  depends_on :java => :optional
+  depends_on :python3 => :optional
+
+  with_python = build.with?("python") || build.with?("python3")
+  pythons = build.with?("python3") ? ["with-python3"] : []
+  depends_on "numpy" => [:recommended] + pythons if with_python
+
+  # Dependencies use fortran, which leads to spurious messages about gcc
```
Comment (ilovezfs) [inline Formula/opencv.rb]: is this still a thing?

File: Formula/opencv.rb [hunk 24]
```diff
@@ -0,0 +1,157 @@
...
+  cxxstdlib_check :skip
+
+  resource "icv-macosx" do
```
Comment (ilovezfs) [inline Formula/opencv.rb]: can we do without this? It says third party

File: Formula/opencv.rb (hunk 13 above)
Comment (iMichka) [inline Formula/opencv.rb]: There is only one PYTHONPATH; and opencv relies on it. Supporting Python 2 and 3 together broke people's build when using Python 3. This is why there is a check for that somewhere in the formula. But I would be willing to remove it and see what happens.

File: Formula/opencv.rb (hunk 15 above)
Comment (iMichka) [inline Formula/opencv.rb]: opencv 3.3.0 fixes this. See PR in homebrew science.

File: Formula/opencv.rb (hunk 13 above)
Comment (ilovezfs) [inline Formula/opencv.rb]: If we can't support both, I think we need to just pick one. This formula shouldn't be built from source as a matter of course.

Comment (iMichka) [inline Formula/opencv.rb]: Let's try to re-enable both and see what happens :)

File: Formula/opencv.rb (hunk 15 above)
Comment (ilovezfs) [inline Formula/opencv.rb]: We should upgrade to that as part of the same PR.

File: Formula/opencv.rb (hunk 18 above)
Comment (ilovezfs) [inline Formula/opencv.rb]: We should upgrade to 3.3.0 as part of the same PR, so it won't be needed.

File: Formula/opencv.rb [hunk 25]
```diff
@@ -0,0 +1,157 @@
...
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "eigen"
+  depends_on "jpeg"
+  depends_on "libpng"
+  depends_on "libtiff"
+  depends_on "openexr"
+  depends_on :python => :recommended unless MacOS.version > :snow_leopard
+  depends_on "ffmpeg" => :optional
```
Comment (ilovezfs) [inline Formula/opencv.rb]: any reason not to make this default on?

File: Formula/opencv.rb [hunk 26]
```diff
@@ -0,0 +1,157 @@
...
+  depends_on "tbb" => :optional
```
Comment (ilovezfs) [inline Formula/opencv.rb]: any reason not to make this default on?

File: Formula/opencv.rb [hunk 27]
```diff
@@ -0,0 +1,157 @@
...
+  depends_on :java => :optional
```
Comment (ilovezfs) [inline Formula/opencv.rb]: any reason not to make this default on?

File: Formula/opencv@2.rb [hunk 28]
```diff
@@ -0,0 +1,98 @@
...
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "eigen"
+  depends_on "jpeg"
+  depends_on "libpng"
+  depends_on "libtiff"
+  depends_on "openexr"
+  depends_on :python => :recommended if MacOS.version <= :snow_leopard
+  depends_on "numpy" => :recommended if build.with? "python"
```
Comment (ilovezfs) [inline Formula/opencv@2.rb]: I think we can just require it.

File: Formula/opencv@2.rb [hunk 29]
```diff
@@ -0,0 +1,98 @@
...
+  depends_on "ffmpeg" => :optional
```
Comment (ilovezfs) [inline Formula/opencv@2.rb]: any reason not to make this default on?

File: Formula/opencv@2.rb [hunk 30]
```diff
@@ -0,0 +1,98 @@
...
+  depends_on "tbb" => :optional
```
Comment (ilovezfs) [inline Formula/opencv@2.rb]: any reason not to make this default on?

File: Formula/opencv@2.rb [hunk 31]
```diff
@@ -0,0 +1,98 @@
...
+  depends_on :java => :optional
```
Comment (ilovezfs) [inline Formula/opencv@2.rb]: any reason not to make this default on?

File: Formula/opencv@2.rb [hunk 32]
```diff
@@ -0,0 +1,98 @@
...
+  def arg_switch(opt)
+    (build.with? opt) ? "ON" : "OFF"
+  end
+
+  def install
+    jpeg = Formula["jpeg"]
+
+    args = std_cmake_args + %W[
+      -DCMAKE_OSX_DEPLOYMENT_TARGET=
+      -DBUILD_JASPER=OFF
```
Comment (ilovezfs) [inline Formula/opencv@2.rb]: does this refer to a vendored jasper?

File: Formula/opencv@2.rb [hunk 33]
```diff
@@ -0,0 +1,98 @@
...
+      -DJPEG_LIBRARY=#{jpeg.opt_lib}/libjpeg.dylib
+    ]
+
+    args << "-DBUILD_opencv_python=" + arg_switch("python")
+    args << "-DBUILD_opencv_java=" + arg_switch("java")
+    args << "-DWITH_FFMPEG=" + arg_switch("ffmpeg")
+    args << "-DWITH_TBB=" + arg_switch("tbb")
+
+    if build.with? "python"
+      py_prefix = `python-config --prefix`.chomp
```
Comment (ilovezfs) [inline Formula/opencv@2.rb]: I don't think this is idiomatic in core.

File: Formula/opencv@2.rb [hunk 34]
```diff
@@ -0,0 +1,98 @@
...
+      args << "-DPYTHON_LIBRARY=#{py_lib}/libpython2.7.dylib"
+      args << "-DPYTHON_INCLUDE_DIR=#{py_prefix}/include/python2.7"
+
+      # Make sure find_program locates system Python
+      # https://github.com/Homebrew/homebrew-science/issues/2302
+      args << "-DCMAKE_PREFIX_PATH=#{py_prefix}"
+    end
+
+    if ENV.compiler == :clang && !build.bottle?
+      args << "-DENABLE_SSSE3=ON" if Hardware::CPU.ssse3?
```
Comment (ilovezfs) [inline Formula/opencv@2.rb]: they can't detect this automatically?

Comment (iMichka) [inline Formula/opencv@2.rb]: as said before (but maybe not same line), only with opencv 3.3.0

File: Formula/opencv.rb (hunk 15 above)
Comment (ilovezfs) [inline Formula/opencv.rb]: If 3.3.0 goes in science first that's OK.

File: Formula/opencv.rb (hunk 18 above)
Comment (ilovezfs) [inline Formula/opencv.rb]: (If 3.3.0 goes in science first that's OK, and then we can revision bump here.)

File: Formula/opencv@2.rb (hunk 33 above)
Comment (fxcoudert) [inline Formula/opencv@2.rb]: This is done in several formulas, including `llvm`

File: Formula/opencv@2.rb (hunk 32 above)
Comment (fxcoudert) [inline Formula/opencv@2.rb]: It does

File: Formula/opencv@2.rb (hunk 29 above)
Comment (fxcoudert) [inline Formula/opencv@2.rb]: Too many deps :(

File: Formula/opencv@2.rb (hunk 32 above)
Comment (ilovezfs) [inline Formula/opencv@2.rb]: Is there a way to say "disable _all_ vendored stuff" so we don't have to play Whac-A-Mole?

Comment (fxcoudert) [inline Formula/opencv@2.rb]: If there is, I haven't found it :(

**Comment by fxcoudert** [comment]
@BrewTestBot test this please

**Comment by fxcoudert** [comment]
@iMichka how do you feel about the current state? are we ready to merge?

**Comment by iMichka** [comment]
I'll build it locally and give it a quick try, I want to see what happens with the python2/python3 import problems reported in science. I have some time now, so I'll report back to you in 1 or 2 hours.

Is there a way to not have the following error:
> W: 107: col 45: Unused block argument - `python`. You can omit all the arguments if you don't care about them.

Can you set the variable to something like `unused` or `_` to prevent the audit to fail (don't know if that works in the ruby world but this is how I do it in Python).

**Comment by iMichka** [comment]
Also we need a PR in science to remove opencv/opencv3. You closed https://github.com/Homebrew/homebrew-science/pull/5996, so either it needs to be re-opened or a new one needs to be created.

**Comment by iMichka** [comment]
Ok so I built the formula as is. `import cv2` works in python3. It is weird that this is called cv2 as the formula is now opencv (3.3.0). But that is probably upstream's decision.

The formula does not build python2 bindings. This is probably due to this line:
> depends_on :python => :recommended if MacOS.version <= :snow_leopard

Is this the wanted behaviour? There is an option to build without python2, but this would only be useful under Snow Leopard or older?

**Comment by fxcoudert** [comment]
@iMichka this is actually working for me:

```
$ python2.7
Python 2.7.13 (default, Jul 18 2017, 09:17:00) 
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import site; site.addsitedir("/usr/local/lib/python2.7/site-packages")
>>> import cv2
>>> print(cv2.__version__)
3.3.0
```

Both python2 and python3 bindings get built.

**Comment by iMichka** [comment]
```
[michkapopoff:~] % python
Python 2.7.13 (default, Jul 18 2017, 09:17:00) 
[GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import cv2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named cv2
>>> import site; site.addsitedir("/usr/local/lib/python2.7/site-packages")
>>> import cv2
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named cv2
```

Python2 does not got built:
https://gist.github.com/a36bcf6c96622544bc3bc240dace386e

Only the python3 .so gets installed
Installing: /usr/local/Cellar/opencv/3.3.0/lib/python3.6/site-packages/cv2.cpython-36m-darwin.so

**Comment by iMichka** [comment]
You can also see this the configuration section:

```
General configuration for OpenCV 3.3.0 =====================================

--   Python 2:
--     Interpreter:                 /usr/bin/python (ver 2.7.10)
--     Libraries:                   NO
--     numpy:                       NO (Python wrappers can not be generated)
--     packages path:               lib/python2.7/site-packages
-- 
--   Python 3:
--     Interpreter:                 /usr/local/opt/python3/bin/python3 (ver 3.6.2)
--     Libraries:                   /usr/local/opt/python3/Frameworks/Python.framework/Versions/3.6/lib/python3.6/config-3.6m-darwin/libpython3.6.dylib (ver 3.6.2)
--     numpy:                       /usr/local/lib/python3.6/site-packages/numpy/core/include (ver 1.13.1)
--     packages path:               lib/python3.6/site-packages
```

It picks up the wrong Python 2 (system python).

**Comment by ilovezfs** [comment]
It should say
```
--   Python 2:
--     Interpreter:                 /usr/bin/python (ver 2.7.10)
--     Libraries:                   /System/Library/Frameworks/Python.framework/Versions/2.7/lib/libpython2.7.dylib (ver 2.7.10)
--     numpy:                       /usr/local/lib/python2.7/site-packages/numpy/core/include (ver 1.13.1)
--     packages path:               lib/python2.7/site-packages
```

**Comment by iMichka** [comment]
`brew doctor` told me that I had an unlinked `numpy` in my Cellar. Cleaning that up and removing all the other numpys installed with pip, the python2 bindings got built and installed. Imports in python 3 and 2 worked correctly. All good for me then.

File: Formula/opencv@2.rb [hunk 35]
```diff
@@ -0,0 +1,87 @@
...
+      #include <opencv/cv.h>
+      #include <iostream>
+      int main() {
+        std::cout << CV_VERSION << std::endl;
+        return 0;
+      }
+    EOS
+    system ENV.cxx, "test.cpp", "-I#{include}", "-L#{lib}", "-o", "test"
+    assert_equal version.to_s, shell_output("./test").strip
+    assert_match version.to_s, shell_output("python -c 'import cv2; print(cv2.__version__)'")
```
Comment (iMichka) [inline Formula/opencv@2.rb]: Can the openc@2 test block be also run for both Python versions with `Language::Python.each_python(build) do |python, _version|`?

Comment (iMichka) [inline Formula/opencv@2.rb]: Err no please disregard that comment, it does not have python3 support anyway :)

**Comment by ilovezfs** [comment]
```
  * W: 108: col 45: Unused block argument - `python`. You can omit all the arguments if you don't care about them.
```

**Review by ilovezfs — APPROVED** [review]

**Review by iMichka — APPROVED** [review]
+1 good to go :)
